src/Server.js

```diff
@@ -59,7 +59,7 @@ function Server(params={ip:createRandomIp(), hostname:""}) {
     this.serversOnNetwork        = [];
 
     //Port information, required for porthacking servers to get admin rights
-    this.numOpenPortsRequired = params.numOpenPortsRequired != null ? params.numOpenPortsRequired : 5;
+    this.numOpenPortsRequired = params.numOpenPortsRequired !== undefined ? params.numOpenPortsRequired : 5;
```

Comment:
From an idiomatic JavaScript perspective, I'd expect to read this as `this.numOpenPortsRequired = params.numOpenPortsRequired || 5;`. But having a private refactoring of a lot of this code, I know that there's too much that is explicitly looking for explicit undefined or null values rather than falsey values.

Reply:
You can't do that, because numOpenPortsRequired can be zero.

Probably the easiest thing is to just have zero be default.